src/libraries/System.IO/tests/StreamReader/StreamReaderTests.cs

```diff
@@ -110,6 +110,53 @@ public async Task ReadToEndAsync()
             Assert.Equal(5000, result.Length);
         }
 
+        [Fact]
+        public async Task ReadToEndAsync_WithCancellationToken()
+        {
+            using var sw = new StreamReader(GetLargeStream());
+            var result = await sw.ReadToEndAsync(default);
+
+            Assert.Equal(5000, result.Length);
+        }
+
+        [Fact]
+        public async Task ReadToEndAsync_WithCanceledCancellationToken()
+        {
+            using var sw = new StreamReader(GetLargeStream());
+            using var cts = new CancellationTokenSource();
+            cts.Cancel();
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await sw.ReadToEndAsync(cts.Token));
+        }
+
+        [Fact]
+        public async Task ReadToEndAsync_WithCancellation()
+        {
+            var path = Path.GetTempFileName();
+            try
+            {
+                // create large (~100MB) file
+                using (var writer = new StreamWriter(path))
+                {
+                    for (var i = 0; i < 1_000_000; i++)
+                        writer.WriteLine("A very large file used for testing StreamReader cancellation. 0123456789012345678901234567890123456789.");
+                }
+
+                using var reader = File.OpenText(path);
+                using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(50));
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await reader.ReadToEndAsync(cts.Token));
+            }
+            finally
+            {
+                try
+                {
+                    File.Delete(path);
+                }
+                catch (Exception)
+                {
+                }
+            }
```

Comment:
I know that this is not obvious and kind of hidden, but this `class` is `partial` and the other definition of it derives from `FileCleanupTestBase` which provides a method for getting temp file path and removing it OOTB

https://github.com/dotnet/runtime/blob/8c188f38bd9b4cca0dda9e7fce63a59294964e60/src/libraries/System.IO/tests/StreamReader/StreamReader.cs#L10

You can also  use `File.WriteAllLines` to simplify the initialization logic.

```suggestion
            string path = GetTestFilePath();
            
            // create large (~100MB) file
            File.WriteAllLines(path, Enumerable.Repeat("A very large file used for testing StreamReader cancellation. 0123456789012345678901234567890123456789.", 1_000_000));

            using StreamReader reader = File.OpenText(path);
            using CancellationTokenSource cts = new (TimeSpan.FromMilliseconds(50));
            
            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await reader.ReadToEndAsync(cts.Token));
```

Reply:
Thanks! I should have tried harder to look for this common functionality.